### ISSUE TITLE
Promote kubescheduler.config.k8s.io to beta

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes/kubernetes#89701]
**What would you like to be added**:
Promote `kubescheduler.config.k8s.io` to Beta by introducing `kubescheduler.config.k8s.io/v1beta1`

/sig scheduling
/wg component-standard

**Why is this needed**:

kube-scheduler has shipped with support for `kubescheduler.config.k8s.io/v1alpha1` since k8s 1.12 and we shipped version `v1alpha2` since k8s 1.18
The most significant changes have been around supporting plugins ([KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/20180409-scheduling-framework.md)) and scheduling profiles ([KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/1451-multi-scheduling-profiles))

We believe it's time to commit to the support of this API by promoting it to beta. This way, more users will be more comfortable incorporating it in their setups.

**Task tracking**:

These are the tasks that need to be finished before 1.19 freeze:

- [x] Write KEP
- [x] Introduce `kubescheduler.config.k8s.io/v1beta1` #91420
- [x] Remove `.bindTimeoutSeconds` once we have a volume binding plugin #83726 #90958
- [x] Merge `Reserve` and `Unreserve` in a single PluginSet #90724

Nice to have:
- [x] Renaming DefaultPodTopologySpread plugin
 #91994
- [x] PostFilter extension point #92012
- [x] Remove case-insensitive parsing of Extenders #65414
- [ ] ~Generate reference docs #88919 kubernetes-sigs/reference-docs#138~
- [x] Add types for scheduling plugin Args #87784
- [ ]  ~Add Feature gates #84922~
- [x] Remove v1alpha1 #89260

**Comment by alculquicondor** [comment]
cc @liggitt @lavalamp @mtaufen @ahg-g @Huang-Wei

**Comment by ahg-g** [comment]
I am not sure " Add types for scheduling plugin Args #87784" is part of this, we can always add those  later, but they are not blockers.

**Comment by alculquicondor** [comment]
I would say having the versioned args is a good sign of maturity of the API. We are also in good track with it, so I wouldn't be worried.

**Comment by cofyc** [comment]
after this is done, do you think it's reasonable to call `Unreserve` if `Reserve` fails?

**Comment by ahg-g** [comment]
> after this is done, do you think it's reasonable to call `Unreserve` if `Reserve` fails?

The scheduler calls `Unreserve` when `Reserve` fails already. 

The impact this change will have is that you will not need to list the plugin as `Unreserve` plugin in component config, `Unreserve` will just be another mandatory function in the `Reserve` plugin, not a standalone plugin.

**Comment by alculquicondor** [comment]
Next step is doing rename of `v1alpha2` into `v1beta1`. I want us to get #91142 merged first.

**Comment by cofyc** [comment]
> The scheduler calls Unreserve when Reserve fails already.

https://github.com/kubernetes/kubernetes/blob/cdbd07f64e2f1392fc9b3c68a6c1e06ee22d6449/pkg/scheduler/scheduler.go#L583-L603

The scheduler run `Unreserve` plugins when `assume` fails but not when `Reserve` plugins fail. Maybe I missed something?

**Comment by ahg-g** [comment]
You are right, I guess what I meant is that RunUnreservePlugins is called on any error happening after RunReservePlugins succeeds.

I believe the assumption was that if RunReservePlugins fails, no reservation happened, and so there is no need to call RunUnreservePlugins. But that is not true because there could be multiple reserve plugins, and even if RunReservePlugins fails, some reserve plugins may have succeeded since it takes a single reserve plugin to fail RunReservePlugins.

I guess we have two options, either call RunUnreservePlugins when RunReservePlugins fails, or call Unreserve inside RunReservePlugins only on the plugins that already succeeded. I think the latter option is better and simplifies Unreserve semantics.

**Comment by cofyc** [comment]
Agree. After we merge `Reserve` and `Unreserve`, the latter option should be easy to implement.

**Comment by damemi** [comment]
@alculquicondor could you update the description? A couple tasks have been done or are in progress that don't seem to be reflected there:
* v1beta1 config is being added with https://github.com/kubernetes/kubernetes/pull/91420
* volumebinding plugin was added in the already-linked PR, is removing `volumeBindSeconds` a separate task that still needs to be done?

**Comment by alculquicondor** [comment]
Updated, thanks

**Comment by Huang-Wei** [comment]
@alculquicondor it'd be nice to link "Scheduler PostFilter API #91547" back here.

**Comment by alculquicondor** [comment]
/close

We successfully completed this. Thanks to all who contributed!
